index.js

```diff
@@ -850,45 +855,58 @@ class Command extends EventEmitter {
 
     // Store the reference to the child process
     this.runningCommand = proc;
+    return Promise.resolve();
   };
 
   /**
+   * @return {Promise}
    * @api private
    */
   _dispatchSubcommand(commandName, operands, unknown) {
     const subCommand = this._findCommand(commandName);
     if (!subCommand) this._helpAndError();
 
     if (subCommand._executableHandler) {
-      this._executeSubCommand(subCommand, operands.concat(unknown));
-    } else {
-      subCommand._parseCommand(operands, unknown);
+      if (subCommand._actionHandler) {
+        const actionPromise = subCommand._actionHandler({ command: this, operands, unknown });
+        if (actionPromise instanceof Promise) {
+          return actionPromise;
+        }
+        return Promise.resolve(actionPromise);
```

Comment:
It would probably be more consistent with JavaScript conventions to check for "thenable" than for a Promise as such, but in fact can just drop the test entirely as `Promise.resolve` handles Promises too.